### PR TITLE
added additional data to internal ManagedCentral

### PR DESCRIPTION
**Pull request by johannes94**
## Description
Changed private/internal APIs openapi specification and regenerated go files to add data to ManagedCentral's in fleet-manager. Added a lot of placeholder and default values in presenter function for a first draft. The fields will be filled with actual data throughout further development of fleet-manager and fleetshard sync.

`api/rhacs/v1/agent_clusters/{id}/centrals`
Previous response:
```
{
  "kind": "ManagedCentralList",
  "items": [
    {
      "id": "cabm59287d5mmi60q43g",
      "kind": "ManagedCentral",
      "metadata": {
        "name": "test132",
        "namespace": "dinosaur-cabm59287d5mmi60q43g",
        "annotations": {
          "mas/id": "cabm59287d5mmi60q43g",
          "mas/placementId": "cabm5aa87d5mmi60q440"
        }
      },
      "spec": {
        "owners": [
          "rh-ee-jmalsam"
        ],
        "endpoint": {
          "host": "test---cabm-----d-mmi--q--g.kubernetes.docker.internal",
          "tls": {}
        },
        "versions": {
          "central": "0.1.0",
          "centralOperator": "0.1.0"
        },
        "deleted": false
      }
    }
  ]
}
```

New response:

```
{
  "kind": "ManagedCentralList",
  "items": [
    {
      "id": "cabi4si87d5kmtcp9bh0",
      "kind": "ManagedCentral",
      "metadata": {
        "name": "test132",
        "namespace": "dinosaur-cabi4si87d5kmtcp9bh0",
        "annotations": {
          "mas/id": "cabi4si87d5kmtcp9bh0",
          "mas/placementId": "cabi51i87d5kmtcp9bhg"
        },
        "deletionTimestamp": ""
      },
      "spec": {
        "owners": [
          "rh-ee-jmalsam"
        ],
        "endpoint": {
          "host": "test---cabi-si--d-kmtcp-bha.kubernetes.docker.internal",
          "tls": {
            "cert": "cert-data",
            "key": "key-data"
          }
        },
        "versions": {
          "central": "0.1.0",
          "centralOperator": "0.1.0"
        },
        "central": {
          "resources": {
            "requests": {
              "cpu": "1000m",
              "memory": "1Gi"
            },
            "limits": {
              "cpu": "1000m",
              "memory": "4Gi"
            }
          }
        },
        "scanner": {
          "analyzer": {
            "scaling": {
              "autoScaling": "enabled",
              "replicas": 1,
              "minReplicas": 1,
              "maxReplicas": 3
            },
            "resources": {
              "requests": {
                "cpu": "500m",
                "memory": "500Mi"
              },
              "limits": {
                "cpu": "2000m",
                "memory": "2500Mi"
              }
            }
          },
          "db": {
            "host": "dbhost.rhacs-psql-instance"
          }
        }
      }
    }
  ]
}
```

## Checklist (Definition of Done)
- [ ] Unit and integration tests added
- [ ] Documentation added if necessary
- [ ] CI and all relevant tests are passing

## Test manual

Run fleet-manager and verify cluster status:
```
make db/teardown db/setup db/migrate
# use config file for your dev cluster
cp dev/config/dataplane-cluster-configuration-dockerdesktop.yaml config/dataplane-cluster-configuration.yaml

make run
# wait until cluster is in db and ready state:
make db/login
select * from clusters;
-[ RECORD 1 ]------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------
id                                   | cabi3ki87d5kmtcp9bg0
created_at                           | 2022-06-01 08:22:42.320843+00
updated_at                           | 2022-06-01 08:22:42.320843+00
deleted_at                           | 
cloud_provider                       | standalone
cluster_id                           | 1234567890abcdef1234567890abcdef
external_id                          | 
multi_az                             | f
region                               | standalone
status                               | ready
identity_provider_id                 | 
cluster_dns                          | kubernetes.docker.internal
provider_type                        | standalone
provider_spec                        | 
cluster_spec                         | 
available_dinosaur_operator_versions | \x5b7b2276657273696f6e223a22302e312e30222c227265616479223a747275652c2263656e7472616c56657273696f6e73223a5b7b2276657273696f6e223a22302e312e30227d5d7d5d
supported_instance_type              | eval,standard
```

Create central resource and verify it transitions to provisioning:
```
function fmcurl {
   resource=${1}
   shift
   curl -v -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/${resource} $@
}

make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development

fmcurl rhacs/v1/centrals\?async=true -XPOST -d '{
    "region": "standalone",
    "cloud_provider": "standalone",
    "name": "test132",
    "host": "needs-to-be-filled-inn",
    "multi_az": true
}'
# return 202

# wait until central is in provisioning status in DB after some time:
select status, id, cluster_id from dinosaur_requests;
# should return something like
-[ RECORD 1 ]--------------------------------
id         | cabm59287d5mmi60q43g
cluster_id | 1234567890abcdef1234567890abcdef
status     | provisioning
```

Query internal GET centrals API:

```
fmcurl rhacs/v1/agent-clusters/1234567890abcdef1234567890abcdef/centrals | jq
{
  "kind": "ManagedCentralList",
  "items": [
    {
      "id": "cabm59287d5mmi60q43g",
      "kind": "ManagedCentral",
      "metadata": {
        "name": "test132",
        "namespace": "dinosaur-cabm59287d5mmi60q43g",
        "annotations": {
          "mas/id": "cabm59287d5mmi60q43g",
          "mas/placementId": "cabm5aa87d5mmi60q440"
        }
      },
      "spec": {
        "owners": [
          "rh-ee-jmalsam"
        ],
        "endpoint": {
          "host": "test---cabm-----d-mmi--q--g.kubernetes.docker.internal",
          "tls": {
            "cert": "cert-data",
            "key": "key-data"
          }
        },
        "versions": {
          "central": "0.1.0",
          "centralOperator": "0.1.0"
        },
        "central": {
          "resources": {
            "requests": {
              "cpu": "1000m",
              "memory": "1Gi"
            },
            "limits": {
              "cpu": "1000m",
              "memory": "4Gi"
            }
          }
        },
        "scanner": {
          "analyzer": {
            "scaling": {
              "autoScaling": "enabled",
              "replicas": 1,
              "minReplicas": 1,
              "maxReplicas": 3
            },
            "resources": {
              "requests": {
                "cpu": "500m",
                "memory": "500Mi"
              },
              "limits": {
                "cpu": "2000m",
                "memory": "2500Mi"
              }
            }
          },
          "db": {
            "host": "dbhost.rhacs-psql-instance"
          }
        }
      }
    }
  ]
}
```

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

**Review by SimonBaeumer — COMMENTED** [review]
Currently we expect that the user can only set a name for the instance from the UI.
All other data is enriched by fleet-manager.

Thus a separation between data sourced by fleet-manager and user makes sense.

**Data from user:**

 - Name
 - Availability Zone / Region (later, we start with `us-east-1` for MVP, EMEA is added later)
 - Additional CAs (later for private PKIs)


**Data from FleetManager:**

 - instance / tenant name
 - TLS certificates
 - Resource limits (for development at first)
 - Subscriptions / Service level
    - Scaling config
    - Resource limits
 - DB configuration for RDS (skip for now, inline comment is okay, add to gaps)
 - DNS name for Central
 - central version
 - deletion timestamp


I would suggest to start with this. Ignoring namespace for now.
After the initial data is implemented and the fleet-shard runtime merged we can start implementing the production configuration implementation.

File: internal/dinosaur/pkg/presenters/manageddinosaur.go [hunk 1]
```diff
@@ -5,6 +5,22 @@ import (
 	v1 "github.com/stackrox/acs-fleet-manager/pkg/api/manageddinosaurs.manageddinosaur.mas/v1"
 )
 
+// TODO(create-ticket): implement configurable central and scanner resources
+const defaultCentralRequestMemory = "1Gi"
```
Comment (kovayur) [inline internal/dinosaur/pkg/presenters/manageddinosaur.go:9]: NIT: combine to a const block?

File: internal/dinosaur/pkg/api/private/api/openapi.yaml [hunk 2]
```diff
@@ -186,14 +186,18 @@ components:
...
         metadata:
           name: example-central
           namespace: example-central-1rfpsqbvq1em2u9u0z54ymjcwac
           annotations:
             mas/id: 1rfpsqbvq1em2u9u0z54ymjcwac
             mas/placementId: ""
+          deletionTimestamp: 2022-06-01T11:04:09Z
         spec:
+          owners:
+          - central-owner1
```
Comment (SimonBaeumer) [inline internal/dinosaur/pkg/api/private/api/openapi.yaml:200]: Why was this changed?

File: internal/dinosaur/pkg/api/private/api/openapi.yaml [hunk 3]
```diff
@@ -202,7 +206,30 @@ components:
           versions:
             central: 2.7.0
             centralOperator: 0.21.2
-          deleted: false
+          central:
+            resources:
+              requests:
+                cpu: 1000m
```
Comment (SimonBaeumer) [inline internal/dinosaur/pkg/api/private/api/openapi.yaml:212]: Can you reduce CPU request to 250m and memory to 250Mi? 
This setting would always consume one CPU core for central

File: internal/dinosaur/pkg/api/private/api/openapi.yaml [hunk 4]
```diff
@@ -202,7 +206,30 @@ components:
...
+          scanner:
+            analyzer:
+              scaling:
+                autoScaling: enabled
+                replicas: 1
+                minReplicas: 1
+                maxReplicas: 3
+              resources:
+                requests:
+                  cpu: 500m
```
Comment (SimonBaeumer) [inline internal/dinosaur/pkg/api/private/api/openapi.yaml:226]: Can you reduce this to 250m?

File: internal/dinosaur/pkg/api/private/api/openapi.yaml [hunk 5]
```diff
@@ -202,7 +206,30 @@ components:
...
+                  memory: 500Mi
```
Comment (SimonBaeumer) [inline internal/dinosaur/pkg/api/private/api/openapi.yaml:227]: Can you reduce this to 100Mi?

File: internal/dinosaur/pkg/api/private/model_managed_central_all_of_spec.go [hunk 6]
```diff
@@ -11,8 +11,10 @@ package private
...
 // ManagedCentralAllOfSpec struct for ManagedCentralAllOfSpec
 type ManagedCentralAllOfSpec struct {
-	Owners   []string                        `json:"owners,omitempty"`
-	Endpoint ManagedCentralAllOfSpecEndpoint `json:"endpoint,omitempty"`
-	Versions ManagedCentralVersions          `json:"versions,omitempty"`
-	Deleted  bool                            `json:"deleted"`
+	Owners        []string                               `json:"owners,omitempty"`
+	Endpoint      ManagedCentralAllOfSpecEndpoint        `json:"endpoint,omitempty"`
+	Versions      ManagedCentralVersions                 `json:"versions,omitempty"`
+	AdditionalCAs []ManagedCentralAllOfSpecAdditionalCAs `json:"additionalCAs,omitempty"`
```
Comment (SimonBaeumer) [inline internal/dinosaur/pkg/api/private/model_managed_central_all_of_spec.go:17]: Let's remove the additionalCA until we need them after MVP.

File: internal/dinosaur/pkg/presenters/manageddinosaur.go (hunk 1 above)
Comment (SimonBaeumer) [inline internal/dinosaur/pkg/presenters/manageddinosaur.go:9]: Use this syntax instead:
```
const (
  defaultCentralRequestMemory = "1Gi"
  defaultCentralRequestCpu = "1000m"
  ...
)
```

File: internal/dinosaur/pkg/api/private/api/openapi.yaml [hunk 7]
```diff
@@ -388,6 +431,48 @@ components:
           type: string
         tls:
           $ref: '#/components/schemas/ManagedCentral_allOf_spec_endpoint_tls'
+    ManagedCentral_allOf_spec_additionalCAs:
```
Comment (SimonBaeumer) [inline internal/dinosaur/pkg/api/private/api/openapi.yaml:434]: Remove additionalCA

**Review by SimonBaeumer — CHANGES_REQUESTED** [review]
Nice!
Some small comments but looks good from my side!

File: internal/dinosaur/pkg/api/private/api/openapi.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (kovayur) [reply internal/dinosaur/pkg/api/private/api/openapi.yaml]: No objections, was just curious.

**Review by kovayur — APPROVED** [review]
Looks good!

**Review by SimonBaeumer — APPROVED** [review]

**Pull request merged**